service/src/main/java/skills/notify/EmailNotifier.groovy

```diff
@@ -109,48 +138,94 @@ class EmailNotifier implements Notifier {
+                    fromEmail = init.fromEmail
+                    formatting = init.formatting
+                }
+
+                def userIds = null
                 try {
-                    sendingService.sendEmail(emailRes.subject, userAttrs.email, emailRes.html, emailRes.plainText, notification.requestedOn)
-                } catch (Throwable t) {
-                    // don't print the same message over and over again
-                    if (!lastErrMsg?.equalsIgnoreCase(t.message)) {
-                        log.error("${prependToLogs}Failed to sent notification with id [${notification.id}] and type [${notification.type}]. Updating notification to retry", t)
-                        lastErrMsg = t.message
+                    userIds = slurper.parseText(notification.userId)
+                } catch (JsonException ex) {
+                    log.warn("user id field [${notification.userId}] was not in the expected json format, this is expected for any notifications that existed prior to 1.6")
+                    userIds = [notification.userId]
+                }
+
+                NotificationEmailBuilder.Res emailRes = notificationEmailBuilderManager.build(notification, formatting)
+                assert notification.userId?.size() > 0
+
+                List<String> failedUserIds = []
+
+                userIds.each {
+                    String email = userAttrs.findEmailByUserId(it)
+                    if (email) {
+                        boolean failed = false;
+                        try {
+                            sendingService.sendEmail(emailRes.subject, email, emailRes.html, emailRes.plainText, notification.requestedOn, senderForBatch, fromEmail)
+                        } catch (Throwable t) {
+                            // don't print the same message over and over again
+                            if (!lastErrMsg?.equalsIgnoreCase(t.message)) {
+                                log.error("${prependToLogs}Failed to send notification with id [${notification.id}] and type [${notification.type}]. Updating notification to retry", t)
+                                lastErrMsg = t.message
+                            }
+                            failed = true;
+                        }
+                        if (!failed) {
+                            count++
+                        } else {
+                            failedUserIds.add(it)
+                            errCount++
+                        }
+                    } else {
+                        log.warn("unable to send notification to recipient [${it}], no email address found")
                     }
-                    notification.failedCount = notification.failedCount + 1
+                }
 
+                if (!failedUserIds) {
+                    removeNotificationImmediately(notification.id)
+                } else {
+                    notification.failedCount = notification.failedCount + 1
```

Comment:
suggesting to add service test for when the email fails for some userIds - verify that the notification is updated with failed ids only